File: LuminaBeauty-final/LuminaBeauty-Model/src/main/java/pe/edu/pucp/luminaBeauty/Model/DetallePedido.java
package pe.edu.pucp.luminaBeauty.Model;
import java.math.BigDecimal;

public class DetallePedido {
    private int id;
    private int cantidad;
    private BigDecimal precioUnitario;
    private BigDecimal subtotal;
    private Pedido pedido; // FK
    private Producto producto; // FK


    public DetallePedido() {
    }

    // Getters y Setters
    public Pedido getPedido() {return pedido;}

    public void setPedido(Pedido pedido) {this.pedido = pedido;}
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getCantidad() { return cantidad; }
    public void setCantidad(int cantidad) { this.cantidad = cantidad; }

    public BigDecimal getPrecioUnitario() { return precioUnitario; }
    public void setPrecioUnitario(BigDecimal precioUnitario) { this.precioUnitario = precioUnitario; }

    public BigDecimal getSubtotal() { return subtotal; }
    public void setSubtotal(BigDecimal subtotal) { this.subtotal = subtotal; }

    public Producto getProducto() { return producto; }
    public void setProducto(Producto producto) { this.producto = producto; }
}

File: lumina_beauty_final_version/lumina_beauty_frontend/LuminaBeauty/Models/DetallePedido.cs
using System.Text.Json.Serialization;

namespace LuminaBeauty.Servicios.Modelo
{
    public class DetallePedido
    {
        [JsonPropertyName("id_detalle_pedido")]
        public int IdDetallePedido { get; set; }

        [JsonPropertyName("producto")]
        public Producto? Producto { get; set; }

        [JsonPropertyName("cantidad")]
        public int Cantidad { get; set; }

        [JsonPropertyName("precio_unitario")]
        public decimal PrecioUnitario { get; set; }

        [JsonPropertyName("subtotal")]
        public decimal Subtotal { get; set; }
    }
}